### PR TITLE
Updates starthistle plant description & stats, adds a seed return harvest, fixes sprites & adds starthistle seeds to the botany vendor contraband

### PR DESCRIPTION
**Pull request by FantasticFwoosh**
:cl: Moonlighting Mac says
add: Starthistles now return seeds when they are harvested, amongst having a overhaul of description, alteration of statistics & a mutational path into harebells. These seeds have been supplied into the hydroponics seed vendor. 
imageadd: Starthistle tray sprites have been restructured & also fixed from a issue of them being broken and not appearing in trays. In addition there are new sprites for its seeds.
/:cl:

It is a glaring hole in the current botany system to have this plants tray sprites never appear due to being left behind codewise in not referencing where its sprites in growing_flower.dmi are held, which has been since corrected & the plant itself revised. 

Though it is nessecarily a unwelcome invading weed; before this change it was only possible you could only shovel the plant out of the tray to great annoyance, so to still establish it as a nuisance plant and give it a silver lining of usefuless, it returns seeds upon harvest like a replica pod & can mutate into the harebell plant which is slightly more useful.

Stocks of this seed have been added to the vendor too, as to mutate into harebells directly rather than wait for weeds to spring up, or to keep around as a convenient tray filler or aesthetic accent, bees will readily pollinate these flowers if you don't actually care about produce returns in their presence.  🐝

### CODE REVIEW

File: code/modules/hydroponics/grown/misc.dm [hunk 1]
```diff
@@ -1,19 +1,32 @@
...
+	mutatelist = list(/obj/item/seeds/harebell)
+
+/obj/item/seeds/starthistle/harvest(mob/user)
+	var/obj/machinery/hydroponics/parent = loc
+	var/seed_count = yield
+	if(prob(getYield() * 20))
+		seed_count++
+		var/output_loc = parent.Adjacent(user) ? user.loc : parent.loc
+		for(var/i in 1 to seed_count)
+			var/obj/item/seeds/starthistle/harvestseeds = src.Copy()
```
Comment (Fox-McCloud) [inline code/modules/hydroponics/grown/misc.dm]: implied `src`.

**Comment by Fox-McCloud** [comment]
I will nearly always support adding more flowers to botany 👍 

Or even more aesthetic plants, in general.

File: code/modules/hydroponics/grown/misc.dm (hunk 1 above)
Comment (FantasticFwoosh) [inline code/modules/hydroponics/grown/misc.dm]: Im not familiar with this, what specifically is the issue, from testing it works out fine. Would you suggest removing it?

Comment (Fox-McCloud) [inline code/modules/hydroponics/grown/misc.dm]: I would. `src` is always implied, and therefore, not needed.

The only time you explicitly need to use `src` (aside form REALLY edge case BYOND bugs that most of us will never see) is if you have a proc that has arguments that are the same as that proc owner has vars:

```
/obj/widget/proc/copy_stuff(A, B, C)
	src.A = A
	src.B = B
	src.C = C
```
That's a case where it's ok to use `src`, but that's really not an argument for using `src` so much as it is avoiding arguments that are identical to vars on the proc owner.

Comment (FantasticFwoosh) [inline code/modules/hydroponics/grown/misc.dm]: I followed your advice by editing it out on my test branch and it worked fine in testing, ill put up the amendment shortly.

**Comment by FantasticFwoosh** [comment]
I can be your worst botany PR foe or your best botany PR friend.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**